docs: add `x/reactions` specs (#958)
desmos-labs/desmos: x/reactions/spec/04-messages.md

---
id: messages
title: Messages
sidebar_label: Messages
slug: messages
---

# Msg Service

## Msg/AddReaction

> With the `MsgAddReaction` a user can react to a post. Below the info needed by the message:

Reviewer: ```suggestion
A post reaction can be added with the `MsgAddReaction`.
```

```js reference
https://github.com/desmos-labs/desmos/blob/020cf82788b667924d0f71f9d8f1fd87efa5b340/proto/desmos/reactions/v1/msgs.proto#L39-L60
```

> The message is expected to fail if any of the following situations occur:
> * The user does not have a profile;
> * The subspace associated with the reaction does not exist;
> * The post to react to does not exist;
> * The post author blocked the user not allowing the interactions with their posts;
> * The user has no permission to react to posts inside the subspace;
> * The reaction's value is not valid;
> * The reaction already exists;
> * The system is unable to retrieve the next reaction ID;
> * The reaction's validation fails.

Reviewer: ```suggestion
It's expected to fail if:
* the user does not have a profile;
* the subspace does not exist;
* the post does not exist;
* the post author has blocked the user within the subspace;
* the user has no permission to react to posts inside the subspace;
* the reaction's value is not valid;
* the reaction already exists.
```

## Msg/RemoveReaction

> 

Reviewer: ```suggestion
A reaction can be removed with the `MsgRemoveReaction`:

```

```js reference
https://github.com/desmos-labs/desmos/blob/020cf82788b667924d0f71f9d8f1fd87efa5b340/proto/desmos/reactions/v1/msgs.proto#L73-L94
```

> The message is expected to fail if any of the following situations occur:
> * The subspace associated with the reaction does not exist;
> * The post to react to does not exist;
> * The reaction does not exist;
> * The user is not equal to the reaction author;
> * The user has no permission to remove reactions within the subspace.

Reviewer: ```suggestion
It's expected to fail if:
* the subspace does not exist;
* the post does not exist;
* the reaction does not exist;
* the user has no permission to remove reactions within the subspace.
```

## Msg/AddRegisteredReaction

> 

Reviewer: ```suggestion
A registered reaction can be added to a subspace with the `MsgAddRegisteredReaction`.

```

```js reference
https://github.com/desmos-labs/desmos/blob/020cf82788b667924d0f71f9d8f1fd87efa5b340/proto/desmos/reactions/v1/msgs.proto#L101-117
```

> The message is expected to fail if any of the following situations occur:
> * The subspace associated with the registered reaction does not exist;
> * The user has no permission to register a reaction within the subspace;
> * The system is unable to retrieve the next registered reaction ID;
> * The reaction validation fails.
> 

Reviewer: ```suggestion
It's expected to fail if:
* the subspace does not exist;
* the user has no permission to register a reaction within the subspace;
* the provided shorthand code is either blank or empty; 
* the provided display value is either blank or empty.
```

## Msg/EditRegisteredReaction

> 

Reviewer: ```suggestion
A registered reaction can be edited with the `MsgEditRegisteredReaction`.

```

```js reference
https://github.com/desmos-labs/desmos/blob/020cf82788b667924d0f71f9d8f1fd87efa5b340/proto/desmos/reactions/v1/msgs.proto#L131-L153
```

> The message is expected to fail if any of the following situations occur:
> * The subspace associated with the registered reaction does not exist;
> * The registered reaction does not exist;
> * The user has no permission to manage registered reactions;
> * The reaction validation fails.

Reviewer: ```suggestion
it's expected to fail if:
* the subspace does not exist;
* the registered reaction does not exist;
* the user has no permission to manage registered reactions;
* the new shorthand code or display value are invalid.
```

## Msg/RemoveRegisteredReaction

> 

Reviewer: ```suggestion
A registered reaction ca be removed with the `MsgRemoveRegisteredReaction`.

```

```js reference
https://github.com/desmos-labs/desmos/blob/020cf82788b667924d0f71f9d8f1fd87efa5b340/proto/desmos/reactions/v1/msgs.proto#L161-L176
```

> The message is expected to fail if any of the following situations occur:
> * The subspace associated with the registered reaction does not exist;
> * The registered reaction does not exist;
> * The user has no permission to manage registered reactions.

Reviewer: ```suggestion
It's expected to fail if:
* the subspace does not exist;
* the registered reaction does not exist;
* the user has no permission to manage registered reactions.
```

---
id: messages
title: Messages
sidebar_label: Messages
slug: messages
---

# Msg Service

## Msg/AddReaction
A post reaction can be added with the `MsgAddReaction`.

```js reference
https://github.com/desmos-labs/desmos/blob/020cf82788b667924d0f71f9d8f1fd87efa5b340/proto/desmos/reactions/v1/msgs.proto#L39-L60
```

It's expected to fail if:
* the user does not have a profile;
* the subspace does not exist;
* the post does not exist;
* the post author has blocked the user within the subspace;
* the user has no permission to react to posts inside the subspace;
* the reaction's value is not valid;
* the reaction already exists.

## Msg/RemoveReaction

> A reaction can be removed with the `MsgRemoveReaction`:

Reviewer: ```suggestion
A reaction can be removed with the `MsgRemoveReaction`.
```
